pkg/pop/poptoken.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package pop
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+)
+
+// PoPKey - generic interface for PoP key properties and methods
+type PoPKey interface {
+	// encryption/signature algo
+	Alg() string
+	// kid
+	KeyID() string
+	// jwk that can be embedded in JWT w/ PoP token's cnf claim
+	JWK() string
+	// https://tools.ietf.org/html/rfc7638 compliant jwk thumbprint
+	JWKThumbprint() string
+	// req_cnf claim that can be included in access token request to AAD
+	ReqCnf() string
+	// sign payload using private key
+	Sign([]byte) ([]byte, error)
+}
+
+// software based pop key implementation of PoPKey
+type swKey struct {
+	key    *rsa.PrivateKey
+	keyID  string
+	jwk    string
+	jwkTP  string
+	reqCnf string
+}
+
+// returns the algorithm used to encrypt/sign the swKey
+func (swk *swKey) Alg() string {
```

Comment:
There is couple function that can be UTs here

Reply:
Mainly to get a good CC

Reply:
+1

Reply:
Added UTs for this file!